hooks/useListInput.tsx

```diff
@@ -0,0 +1,25 @@
+import { Autocomplete, TextField } from "@mui/material";
+import { uniq } from "lodash";
+import { useState, useMemo } from "react";
+
+export const useListInput = <TBranch extends Record<string, string>>(branches: TBranch[], key: keyof TBranch, label: string ) => {
```

Comment:
This component can be more general.
It only works with branches right now.

At the bare minimum rename it to `useBranchPropertyListInput`

Reply:
You tried to make the name more generic, it can run on any list of objects.